### PR TITLE
WIP: Ci 

### PR DESCRIPTION
**Pull request by hfattahi**
This PR is an attempt to fix the build on circle-ci. 
On linux machine I got same error message as reported in circle-ci, which was a complain about not being able to find OpenMP. 
Additionally, for me cython from conda was not detected unless I follow the same change in  https://github.com/isce-framework/fringe/pull/20. Simply adding ```-DCMAKE_PREFIX_PATH=$CONDA_PREFIX``` to the cmake command was not helpful. Happy to hear better solutions. 

I have narrowed the python version in the environment yaml file. This setup was successful locally. Will relax the python version once I see the ci builds sucessfully.

### CODE REVIEW

**Review by piyushrpt — APPROVED** [review]

**Comment by yunjunz** [comment]
I would suggest renaming `requirements_linux.yml` to `environment_linux.yml`.

What's the rationale for removing the `requirements_linux.txt` file? It provides an alternative choice and can not be replaced the yml file, which constrains the user to create a new environment.

**Comment by hfattahi** [comment]
> I would suggest renaming `requirements_linux.yml` to `environment_linux.yml` for the convention consistency.
> 
> What's the rationale for removing the `requirements_linux.txt` file? It provides an alternative choice and can not be replaced the yml file, which constrains the user to create a new environment.

Renaming sounds good. Will do.

I don't think if we should keep multiple environment files floating around. Specially that it did not work on circle-ci and did not work for me locally. I guess a user can always copy those few lines to a new environment files if they want to.

**Review by yunjunz — APPROVED** [review]
LGTM.

File: cmake/FindCython.cmake [hunk 1]
```diff
@@ -28,7 +28,7 @@ find_package(Python 3.6)
 if(Python_Interpreter_FOUND)
   get_filename_component( _python_path ${Python_EXECUTABLE} PATH )
   find_program( CYTHON_EXECUTABLE
-      NAMES cython${Python_VERSION_MAJOR} cython-${Python_VERSION_MAJOR}.${Python_VERSION_MINOR} cython cython.bat
+	  NAMES cython cython.bat cython${Python_VERSION_MAJOR} cython-${Python_VERSION_MAJOR}.${Python_VERSION_MINOR} cython3
```
Comment (rtburns-jpl) [inline cmake/FindCython.cmake:31]: I wouldn't change the order of these, cython3 should always be preferred to plain cython (which might be python2)

Comment (hfattahi) [reply cmake/FindCython.cmake:31]: Thanks @rtburns-jpl. It looks like the problem is that on the machine I'm trying (dgx) there is a pre-installed ```cython3``` at ```/usr/local/bin/cython3```. The cython in conda bin directory is just called ```cython``` not ```cython3```. So I guess that is the issue why the one from system is chosen if I don't change the order. Any suggestion to force the cython from conda even though it is called cython and not cython3?

Comment (rtburns-jpl) [reply cmake/FindCython.cmake:31]: I think you can explicitly override with e.g. `-DCYTHON_EXECUTABLE=$CONDA_PREFIX/bin/cython`

We could also try `${Python_EXECUTABLE} -c cython` like we do in isce3, which should more reliably pick up the right cython installation

Comment (hfattahi) [reply cmake/FindCython.cmake]: ```-DCYTHON_EXECUTABLE=$CONDA_PREFIX/bin/cython``` worked fine. So I reverted the order as you suggested.

**Review by rtburns-jpl — APPROVED** [review]

**Pull request merged**